monitor/filter_exception_decoder.py

```
@@ -12,57 +12,157 @@
-    load_build_metadata,
-)
-from platformio.package.manager.tool import ToolPackageManager
+try:
+    from elftools.elf.elffile import ELFFile
+    from elftools.elf.constants import SH_FLAGS
+
+    HAS_PYELFTOOLS = True
+except ImportError:
+    HAS_PYELFTOOLS = False
+
+env = DefaultEnvironment()
```

Ruff (0.15.2)

[error] Undefined name `DefaultEnvironment`

(F821)